customize_image.sh

```diff
@@ -314,27 +314,20 @@ if [ -e "$IMAGEDIR/etc/init.d/apache2" ] ; then
 fi
 
 if [ "$MINIMG" != "1" ]; then
-	# fake a revpi procfs ... otherwise NodeRed installer will get confused
-	for procfile in meminfo cpuinfo; do
-		touch "$IMAGEDIR/proc/$procfile"
-		mount -o ro,bind "$BAKERYDIR/templates/fake-procfs/$procfile" "$IMAGEDIR/proc/$procfile"
-	done
-
-	# install nodejs and nodered with an install script and revpi-nodes from npm repository
-	NODEREDSCRIPT="/tmp/update-nodejs-and-nodered.sh"
-	/usr/bin/curl -sL \
-		https://raw.githubusercontent.com/node-red/linux-installers/master/deb/update-nodejs-and-nodered\
-		--output "$IMAGEDIR/$NODEREDSCRIPT"
-	chmod 755 "$IMAGEDIR/$NODEREDSCRIPT"
+	# Use NodeJS LTS 18 sources
+	cp "$BAKERYDIR/templates/nodered/nodesource.gpg" "$IMAGEDIR/etc/apt/trusted.gpg.d"
+	cp "$BAKERYDIR/templates/nodered/nodesource.list" "$IMAGEDIR/etc/apt/sources.list.d"
+
+	# Install nodejs 18 including npm
+	chroot "$IMAGEDIR" apt-get update
+	chroot "$IMAGEDIR" apt-get -y install nodejs
+
+	# Install node-red via npm as explained in the node-red documentation
 	NODERED_VER="3.0.2"
-	chroot "$IMAGEDIR" /usr/bin/sudo -u pi $NODEREDSCRIPT --confirm-install --confirm-pi --no-init --nodered-version="$NODERED_VER"
-	rm "$IMAGEDIR/$NODEREDSCRIPT"
-
-	# remove fake procfs after NodeRed setup
-	for procfile in meminfo cpuinfo; do
-		umount "$IMAGEDIR/proc/$procfile"
-		rm "$IMAGEDIR/proc/$procfile"
-	done
+	chroot "$IMAGEDIR" npm install -g --unsafe-perm node-red@${NODERED_VER} || true
```

Comment:
Shouldn't building the image fail if installing `node-red` fails?

Reply:
Unfortunately, we have a problem with npm in a change root environment. There is always an error code! = 0. Maybe you could do a test afterwards to see if certain files are installed? Don't think the idea is bad at all.

Would you do that?

Reply:
I think I'm the wrong person to debug NodeJS related things. @LukasZech can you look into this?